tests/e2e/cucumber/steps/ui/links.ts

```diff
@@ -158,7 +214,26 @@ Then(
     const linkObject = new objects.applicationFiles.Link({ page })
     const actualErrorMessage = await linkObject.checkErrorMessage()
     expect(actualErrorMessage).toBe(errorMessage)
-    await linkObject.clickOnCancelButton()
+  }
+)
+
+Then(
+  '{string} should get the copied password {string}',
+  async function (this: World, stepUser: any, expectedPassword: string): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const linkObject = new objects.applicationFiles.Link({ page })
+    const copiedPassword = await linkObject.getCopiedPassword()
+    expect(copiedPassword).toBe(expectedPassword)
+  }
+)
+
+Then(
+  '{string} should see random generated password',
+  async function (this: World, stepUser: any): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const linkObject = new objects.applicationFiles.Link({ page })
+    const generatedPassword = await linkObject.getEnteredPassword()
+    await expect(generatedPassword).toMatch(/^[A-Za-z0-9\s\S]{12}$/)
```

Comment:
may be declare a sperate variable for this regex. eg like `expectedRegexForGeneratedPassword` and then assert?